app/javascript/components/lease_ups/LeaseUpApplicationsPage.js

```diff
@@ -190,36 +204,71 @@ const LeaseUpApplicationsPage = () => {
       })
   }
 
-  const updateStatusModal = (values) => {
-    setStateWithPrev((prevState) => ({
-      statusModal: {
-        ...clone(prevState.statusModal),
-        ...values
-      }
-    }))
+  const handleCloseStatusModal = () => {
+    updateStatusModal({
+      isOpen: false,
+      showAlert: false,
+      bulkUpdateCount: null,
+      applicationId: null
+    })
+  }
+  const handleLeaseUpStatusChange = (value, applicationId) => {
+    const appsToUpdate = applicationId
+      ? [applicationId]
+      : Object.keys(bulkCheckboxesState).filter((id) => bulkCheckboxesState[id])
```

Comment:
can use entries instead

```suggestion
      : Object.entries(bulkCheckboxesState).filter(([id, checked]) => checked)
```